### DISCUSSION TITLE
Reverse items on screen

### DISCUSSION THREAD

**Discussion by c-harding** [rxhanson/Rectangle#518] [Ideas]
A useful command would be to reverse all items on the current screen, so that windows on the left are moved to the right, and vice versa.

This would be implemented by setting (pseudocode) `window.positionRelativeToScreen.left = window.screen.width -window.positionRelativeToScreen.left - window.width`. The other metadata may also have to be adapted, so that commands like Smaller work correct. It would be acceptable to only mirror the windows that have been positioned by Rectangle, if access to the other windows is not possible.

I would be happy to implement this if it is a feature you’d like.

**Comment by rxhanson** [comment]
Sure, sounds good to me. This was requested a long time ago and I kind of forgot about it. 

Take a look at the Todo Mode stuff in code for a reference on how to manipulate all windows.

**Reply by c-harding** [reply]
Cheers, that looks very useful, thank you. Quick question, why is Defaults.todo an OptionalBoolDefault rather than a BoolDefault? I can’t see any difference between it being false and unset.

**Reply by rxhanson** [reply]
The only difference is future-proofing (perhaps unnecessary). If there were ever a scenario that I wanted todo to be enabled by default for all users in a new release, then I would have to check versions on launch to tell if an update took place, and then set todo to true. The fact that Bool defaults default to false can be really annoying, so I typically just use OptionalBoolDefaults in most places unless a BoolDefault is a really obvious fit.
